src/wallet/bdk.rs

```diff
@@ -103,11 +103,40 @@ impl KeychainKind {
         match self {
             Self::External => 0,
             Self::Internal => 1,
-            Self::Fidelity => 2,
-            Self::SwapCoin => 3,
-            Self::Contract => 4,
+            Self::Fidelity { .. } => 2,
+            Self::SwapCoin { .. } => 3,
+            Self::Contract { .. } => 4,
```

Comment:
This means for all swap/contract coins index number will be 3/4? 

Which is fine. Then we can add another path `2/index` at that keychain for new pubkeys for each new swaps.

Or else you can do `Self::Fidelity{idx} => 2 + ids;`. And then you would need to handle over lap cases. 2 + 1 = 3, but that index is already used by other chain, so always check of there are overlaping indexes.

We can do which ever you prefer. 

I would like the first one, as that allows us to do `m/84'/0'/0'/3` as a watch-only Xpub for all possible swap coins. Then individual swap pubkeys would be at `m/84'/0'/0'/3/1, 2, ...`. Also overlaps don't occur as each type has its own keychain.

Reply:
ACK